crates/bevy_asset/src/path.rs

```diff
@@ -659,12 +672,80 @@ pub(crate) fn normalize_path(path: &Path) -> PathBuf {
     result_path
 }
 
+/// Normalizes the path by collapsing all occurrences of '.' and '..' dot-segments where possible
+/// as per [RFC 1808](https://datatracker.ietf.org/doc/html/rfc1808)
+/// Returns a borrowed path if no normalization was necessary, otherwise returns an owned normalized path.
+pub(crate) fn maybe_normalize_path<'a, P: AsRef<Path> + Into<CowArc<'a, Path>> + 'a>(
```

Comment:
Could we fold this in with `normalize_path` above? Having two impls of the same thing is not great. Besides, the existing callsite could benefit from your improvements here. I think we can just return `Cow<'a, Path>`, and then convert that into a `CowArc` later (or in the case of the existing `normalize_path`, just don't set the result_path at all.